### PR TITLE
1523 Offline UI - connecting for the first time

### LINKED ISSUE #1523: Need offline UI: Connection interrupted
Original issue is #135, which has been split into multiple issues. 

We need UI specs and implementation for:
- When the bot connection has been interrupted.
  - Immediate interruption (including immediate reconnect?)
  - Non-immediate interruption (up to 2 mins)

### What we need
- [x] Design specs
- [x] Acceptance criteria
- [x] T-shirt sizing

-	Grey text
-	With spinner (spinner will be grey, small, size of icon in previous Offline UI items)
-	Text proposal: “Network interruption occurred. Reconnecting…”
-	This will show up as soon as an interruption has occurred, and either disappear (reconnect successful) or switch to “Web Chat is taking longer than usual to connect” after 15 seconds. (Will eventually turn to “Connection failed” if it fails completely)

### LINKED ISSUE #1856: Customize Web Chat with Reaction Buttons does not work
Hi,

I am trying to get this sample. I have setup in my local and followed the steps mentioned here: https://github.com/Microsoft/BotFramework-WebChat/tree/master/samples/09.customization-reaction-buttons/ to set it up and run. I get below error in my localhost in Chrome browser.

image

Same error with the demo link: https://microsoft.github.io/BotFramework-WebChat/09.customization-reaction-buttons/

Error message in Edge is as below.

image

Any help with this please?

### PR DESCRIPTION
**Pull request by corinagum**
Resolves #1523 
Fix #1856 

This now brings 'Connecting...' text to Web Chat when connecting for the first time. 

Scenario 1: User connects to Web Chat for the first time, and 'Connecting...' disappears after connection via DirectLine is made.
ezgif-5-a2cbe28dddf0

Scenario 2: User connects to Web Chat for the first time, and 'Connecting...' shows until it changes to 'Taking longer than usual to connect'
Media2

(difference in quality is due to making the gifs on different websites)

Notes 
- Because of the possibility of 'Connecting...' showing extremely briefly, I added a 400ms pause that keeps the text on screen for a little bit after connection is made.

### CODE REVIEW

File: packages/component/src/Styles/StyleSet/ConnectivityNotification.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export default function ({
+  connectivityMarginLeftRight,
+  connectivityMarginTopBottom,
+  connectivityTextSize,
+  notificationText,
+  primaryFont
+
```
Comment (tonyanziano) [inline packages/component/src/Styles/StyleSet/ConnectivityNotification.js]: whitespace

File: packages/core/src/reducers/connectivityStatus.js [hunk 2]
```diff
@@ -2,33 +2,66 @@ import {
...
+      }
       break;
 
     case CONNECT_REJECTED:
-      state = 'error';
+      state = {
+        ...state,
+        status: 'error',
+        timeoutCompleted: false
+      }
```
Comment (tonyanziano) [inline packages/core/src/reducers/connectivityStatus.js]: semicolon `};`

File: packages/core/src/reducers/connectivityStatus.js [hunk 3]
```diff
@@ -2,33 +2,66 @@ import {
...
       break;
 
     case CONNECT_STILL_PENDING:
-      state = 'connectingslow';
+      state = {
+        ...state,
+        status: 'connectingslow',
+        timeoutCompleted: false
+      }
```
Comment (tonyanziano) [inline packages/core/src/reducers/connectivityStatus.js]: semicolon `};`

File: packages/core/src/reducers/connectivityStatus.js [hunk 4]
```diff
@@ -2,33 +2,66 @@ import {
...
       break;
 
     case DISCONNECT_FULFILLED:
-      state = 'notconnected';
+      state = {
+        ...state,
+        status: 'notconnected',
+        timeoutCompleted: false
+      }
```
Comment (tonyanziano) [inline packages/core/src/reducers/connectivityStatus.js]: semicolon `};`

File: packages/core/src/reducers/connectivityStatus.js [hunk 5]
```diff
@@ -2,33 +2,66 @@ import {
...
+      break;
+
+    case CONNECT_TIMEOUT_COMPLETE:
+      state = {
+        ...state,
+        timeoutCompleted: true
+      }
```
Comment (tonyanziano) [inline packages/core/src/reducers/connectivityStatus.js]: semicolon `};`

**Comment by Souvik04** [comment]
I can see the connecting loader appears briefly. I have a scenario where I use a back channel activity to post data to user as a welcome message. Sometimes, the message might take a while to appear in the conversation window. Is there a way to wait for the back channel post back welcome message and show the connecting loader until then?

**Pull request merged**